psl-core/src/main/java/org/linqs/psl/application/inference/online/OnlineInference.java

```diff
@@ -91,6 +108,46 @@ protected void executeAction(OnlineMessage action) {
         server.onActionExecution(action, new ActionStatus(action, true, response));
     }
 
+    protected String doAddAtom(AddAtom action) {
```

Comment:
The semantics around existing atoms needs to be written in a comment somewhere (just something like: "if the atom already exists as READ or WRITE, then ...").
The think the class comment for the AddAtom AtomAction would be the best.